### PR TITLE
chore: add custom documentation to the README

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +116,247 @@ has instructions for running the samples.
 | --------------------------- | --------------------------------- |
 | Examples | [source code](https://github.com/googleapis/nodejs-error-reporting/blob/master/samples/snippets.js) |
 
+## Configuration
+
+The following code snippet lists all available configuration options.  All configuration options are optional.
```
Comment (ofrobots) [inline README.md]: `all` is probably too strong. I think there are other options that the `Service` parent class accepts that we simply pass on.

Comment (DominicKramer) [reply README.md]: Good catch.

**Comment by JustinBeckwith** [comment]
It's all good.  Thanks for sticking with me on this one.  We can re-eval this when we need to do the fancy README gen tool.

**Comment by DominicKramer** [comment]
@JustinBeckwith Awesome thanks.

**Pull request merged**